crates/store/src/state.rs

```diff
@@ -96,17 +96,114 @@ pub struct TransactionInputs {
     pub found_unauthenticated_notes: BTreeSet<NoteId>,
 }
 
+#[derive(Debug)]
+pub struct Blockchain(Mmr);
+
+impl Blockchain {
+    pub fn get(&self) -> &Mmr {
+        &self.0
+    }
+    pub fn get_mut(&mut self) -> &mut Mmr {
+        &mut self.0
+    }
```

Comment:
We shouldn't need these `get` or `get_mut` methods. We should expose anything that is needed from the `Blockchain` as dedicated methods and with the chain-specific types, e.g.:
- `Blockchain::peaks(&self) -> ...`
- `Blockchain::peaks_at(&self, block_num: BlockNumber) -> ...`
- `Blockchain::add(&mut self, block_hash: Digest) -> ...`
- `Blockchain::open(&self, block_num: BlockNumber) -> ...`